tests/dev/test_update_mlflow_versions.py

```diff
@@ -0,0 +1,116 @@
+import difflib
+import os
+import re
+from pathlib import Path
+
+import pytest
+from packaging.version import Version
+
+from dev.update_mlflow_versions import (
+    get_current_py_version,
+    replace_java,
+    replace_java_pom_xml,
+    replace_js,
+    replace_pyproject_toml,
+    replace_python,
+    replace_r,
+)
+
+# { filename: expected lines changed }
+_JAVA_FILES = {
+    "mlflow/java/scoring/src/main/java/org/mlflow/sagemaker/ScoringServer.java": [175],
+    "mlflow/java/scoring/src/test/java/org/mlflow/ScoringServerTest.java": [81],
+}
+
+_JAVA_XML_FILES = {
+    "mlflow/java/pom.xml": [6, 62],
+    "mlflow/java/scoring/pom.xml": [8],
+    "mlflow/java/client/pom.xml": [8],
+    "mlflow/java/spark/pom.xml": [4, 19],
+}
+
+_JS_FILES = {
+    "mlflow/server/js/src/common/constants.tsx": [12],
+}
+
+_PYTHON_FILES = {
+    "mlflow/version.py": [4],
+}
+
+_PYPROJECT_TOML_FILES = {
+    "pyproject.skinny.toml": [7],
+    "pyproject.toml": [7],
+}
+
+_R_FILES = {
+    "mlflow/R/mlflow/DESCRIPTION": [4],
+}
+
+_DIFF_REGEX = re.compile(r"--- (\d+) ----")
+
+old_version = Version(get_current_py_version())
+_NEW_PY_VERSION = f"{old_version.major}.{old_version.minor}.{old_version.micro + 1}"
+
+
+def copy_and_run_change_func(tmp_path, paths_to_copy, replace_func, new_version):
+    for path in paths_to_copy:
+        copy_path = tmp_path / path
+        copy_path.parent.mkdir(parents=True, exist_ok=True)
+        copy_path.write_text(path.read_text())
+
+    cwd = os.getcwd()
+    os.chdir(tmp_path)
```

Comment:
`monkeypatch.chdir` should be used when changing the working directory in tests.

Reply:
/ai plz explain why